textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -32,78 +32,86 @@
-                luceneFieldType.setStored(true);
-                luceneFieldType.setStoreTermVectors( true );
-                luceneFieldType.setStoreTermVectorOffsets( true );
-                luceneFieldType.setStoreTermVectorPayloads( true );
-                luceneFieldType.setStoreTermVectorPositions( true );
-                luceneFieldType.setTokenized( true );
-
-                luceneField = new org.apache.lucene.document.Field(
-                        fieldName,(String) fieldValue,luceneFieldType);
-
-                break;
+        switch (fieldType) {
+        case STRING:
+            luceneField = new org.apache.lucene.document.StringField(fieldName, (String) fieldValue, Store.YES);
+            break;
+        case INTEGER:
+            luceneField = new org.apache.lucene.document.IntField(fieldName, (Integer) fieldValue, Store.YES);
+            break;
+        case DOUBLE:
+            double value = (Double) fieldValue;
+            luceneField = new org.apache.lucene.document.DoubleField(fieldName, value, Store.YES);
+            break;
+        case DATE:
+            String dateString = DateTools.dateToString((Date) fieldValue, Resolution.MILLISECOND);
+            luceneField = new org.apache.lucene.document.StringField(fieldName, dateString, Store.YES);
+            break;
+        case TEXT:
+            // By default we enable positional indexing in Lucene so that we can
+            // return
+            // information about character offsets and token offsets
+            org.apache.lucene.document.FieldType luceneFieldType = new org.apache.lucene.document.FieldType();
+            luceneFieldType.setIndexOptions(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS);
+            luceneFieldType.setStored(true);
+            luceneFieldType.setStoreTermVectors(true);
+            luceneFieldType.setStoreTermVectorOffsets(true);
+            luceneFieldType.setStoreTermVectorPayloads(true);
+            luceneFieldType.setStoreTermVectorPositions(true);
+            luceneFieldType.setTokenized(true);
+
+            luceneField = new org.apache.lucene.document.Field(fieldName, (String) fieldValue, luceneFieldType);
+
+            break;
 
         }
         return luceneField;
     }
+
     /**
-     * @about Creating a new span tuple from span schema, field list 
+     * @about Creating a new span tuple from span schema, field list
      */
-    public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList, Schema spanSchema) {
+    public static ITuple getSpanTuple(List<IField> fieldList, List<Span> spanList, Schema spanSchema,
+            boolean removeLastField) {
         IField spanListField = new ListField<Span>(new ArrayList<>(spanList));
         List<IField> fieldListDuplicate = new ArrayList<>(fieldList);
+        // If SourceOperator is of type Keyword, KeywordMatcher also includes a
+        // span which should be discarded by dictionary matcher.

```

Comment:
Then I wonder why we only remove the last field.  What if the query has three keyword "new york city"?  In this case should we remove the last two spans instead of the last span?